### PR TITLE
Avoid fetching all cards all the time

### PR DESCRIPTION
**Pull request by kupkovski**
**What is the content type?**

  - Feature

**Why is this change necessary?**

  - Today, every time a card is created, updated, or deleted on the Pipe, or the Pipe Filter changes, the calendar refetches all cards. This change is needed to avoid this behavior, fetching only the affected card.

  - Plus we fix a problem of keeping the selected filter from the pipe.

**How does it address the issue?**

  - When a card is created or updated, we fetch this card from the database and update Apollo's store with that.
  - When a card is deleted, we just remove it from the store.
  - When filter changes we refetch it based on the current Filter.

**What side effects does this change have?**

  - None

**Link to a Pipefy Card?**

  - [Keep the filter options when change calendar views](https://app.pipefy.com/pipes/296086#cards/6319427)

### CODE REVIEW

File: src/components/calendar.jsx [hunk 1]
```diff
@@ -8,43 +9,153 @@ import 'react-big-calendar/lib/css/react-big-calendar.css';
...
+  filter: { labelIds: number[], assigneeIds: number[], title: string },
 };
 
 type State = {
-  cards: Array<Card>,
-  currentDate: Date,
+  currentDate: Date | string,
   currentView: string,
-  loading: boolean,
+  filter: any,
```
Comment (AZanellato) [inline src/components/calendar.jsx]: Could we change this to a actual type? :innocent:

File: src/containers/calendar.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const getFilter = filter =>
+  mountFilters(
+    startDateByView(defaultDate, defaultView),
+    endDateByView(defaultDate, defaultView),
+    filter
+  );
+
+const cardSearchQueryOptions = {
+  options: ({ pipefy: { organizationId, app: { pipeId } }, filter }) => ({
+    variables: {
```
Comment (AZanellato) [inline src/containers/calendar.js]: Maybe we could replace the `filter: getFilter(filter)` with something like this?
```javascript
filterWithoutDate}) => {
const filter = getFilter(filterWithoutDate)
return { variables: { ... } }
```

File: src/components/calendar.jsx [hunk 3]
```diff
@@ -8,43 +9,153 @@ import 'react-big-calendar/lib/css/react-big-calendar.css';
...
+    currentDate: string | Date,
+    { internalId }: { internalId: number }
+  ) => {
+    const { client, data: { variables } } = this.props;
+    updateStore(client, variables, internalId, cachedCards => cachedCards);
+  };
+
+  addCard = (
+    currentView: string,
+    currentDate: string | Date,
```
Comment (ricalvares) [inline src/components/calendar.jsx]: This parameter isn't being used, could you remove it, please?

File: src/components/calendar.jsx [hunk 4]
```diff
@@ -8,43 +9,153 @@ import 'react-big-calendar/lib/css/react-big-calendar.css';
...
+          ...cardSearch,
+          cards: cachedCards.filter(e => e.id !== `${internalId}`),
+        },
+      },
+    });
+  };
+
+  updateCard = (
+    currentView: string,
+    currentDate: string | Date,
```
Comment (ricalvares) [inline src/components/calendar.jsx]: This parameter isn't being used, could you remove it, please?

**Pull request merged**